### PR TITLE
fcitx5-anthy: init at 5.0.13

### PR DESCRIPTION
**Pull request by ElnuDev**
###### Description of changes

Added the anthy wrapper for fcitx5, [fcitx5-anthy](https://github.com/fcitx/fcitx5-anthy). I've also added myself to the maintainers list.

This is my first time contributing to nixpkgs, so I hope I've done everything alright! I based the derivation on fcitx-anthy, the wrapper for the legacy version on fcitx.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4176,6 +4176,12 @@
     githubId = 5300871;
     name = "Leon Kowarschick";
   };
+  elnudev = {
+    email = "elnu@elnu.com";
+    github = "elnudev";
+    githubId = 9874955;
+    name = "Elnu";
+  };
```
Comment (lilyinstarlight) [inline maintainers/maintainer-list.nix:4184]: This is buried deep within the manual (and we should maybe add it to CONTRIBUTING.md too...), but additions to `maintainers/maintainter-list.nix` are supposed to be in their own commits titled "maintainers: add [name]": https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers

File: pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ lib, stdenv, fetchurl, cmake, extra-cmake-modules, pkg-config, fcitx5, anthy, gettext }:
+
+stdenv.mkDerivation rec {
+  pname = "fcitx5-anthy";
+  version = "5.0.13";
+
+  src = fetchurl {
+    url = "https://download.fcitx-im.org/fcitx5/fcitx5-anthy/${pname}-${version}.tar.xz";
+    sha256 = "EGpyYiNalm4En3Oa++hyOSPLhjbkVcDyZt3LqamfSOI=";
```
Comment (lilyinstarlight) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix:9]: This looks like an SRI hash but is missing the `sha256-` prefix. Is this hash right?

File: pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  nativeBuildInputs = [ cmake extra-cmake-modules pkg-config ];
+  buildInputs = [ fcitx5 anthy gettext ];
+
+  preinstall = ''
```
Comment (lilyinstarlight) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix:15]:
```suggestion
  preInstall = ''
```

This should be `preInstall`, with a capital `i`

File: pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{ lib, stdenv, fetchurl, cmake, extra-cmake-modules, pkg-config, fcitx5, anthy, gettext }:
+
+stdenv.mkDerivation rec {
```
Comment (lilyinstarlight) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix:3]: Make sure to add new packages to `pkgs/top-level/all-packages.nix` with a statement like `fcitx5-anthy = callPackage ../tools/inputmethods/fcitx5/fcitx5-anthy.nix { };` so they can be reachable as `fcitx5-anthy`

**Review by lilyinstarlight — CHANGES_REQUESTED** [review]
This is a promising start, but needs a few things before it'll build and work

You can try running `nix-shell -p nixpkgs-review --run 'nixpkgs-review rev HEAD'` from your nixpkgs checkout to test whether it successfully finds and builds the package or not

File: pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix (hunk 2 above)
Comment (ElnuDev) [reply pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix:9]: Yes, this hash is right, I don't think `fetchurl` cares whether you have the `sha256-` prefix or not? It built just fine on my system when using this derivation in my config. I'll add the prefix.

Comment (lilyinstarlight) [reply pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix]: Ah you specified `sha256 =` instead of `hash =`. I should have looked more closely :)

You can put bare, base64 hashes in `sha256` but it is best to leave the prefix so someone else who comes along isn't momentarily puzzled too

**Review by lilyinstarlight — APPROVED** [review]
Awesome! This looks to me now and builds locally (I didn't actually try it in fcitx5 though)

Result of `nixpkgs-review pr 209638` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fcitx5-anthy</li>
  </ul>
</details>

**Comment by ElnuDev** [comment]
Thank you for the review! I have yet to set up Nix/NixOS on my main machine, and running `nix-shell -p nixpkgs-review --run 'nixpkgs-review rev HEAD'` completely froze up my older laptop so I didn't manage to run the review command, sorry about that.

**Comment by lilyinstarlight** [comment]
> Thank you for the review! I have yet to set up Nix/NixOS on my main machine, and running `nix-shell -p nixpkgs-review --run 'nixpkgs-review rev HEAD'` completely froze up my older laptop so I didn't manage to run the review command, sorry about that.

No worries! Sounds like your older laptop doesn't have enough RAM for the expensive eval it does to compare which packages have changed (which I mean, my laptop until a few weeks ago would require me to close all web browsers and apps and even then still had to use some swap for the nixpkgs-review evals to complete)

**Comment by NickCao** [comment]
@ElnuDev did you test if you can input Japanese with this installed, if so we can merge this. cc @Vonfry

File: pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    url = "https://download.fcitx-im.org/fcitx5/fcitx5-anthy/${pname}-${version}.tar.xz";
+    sha256 = "sha256-EGpyYiNalm4En3Oa++hyOSPLhjbkVcDyZt3LqamfSOI=";
+  };
+
+  nativeBuildInputs = [ cmake extra-cmake-modules pkg-config ];
+  buildInputs = [ fcitx5 anthy gettext ];
+
+  preInstall = ''
+    substituteInPlace src/cmake_install.cmake \
+      --replace ${fcitx5} $out
```
[the remark this answers is not in the document]
Comment (ElnuDev) [reply pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix:17]: It builds and works just fine both with and without it for me. It was there to begin with because I was initially basing this on [the derivation of fcitx-anthy](https://github.com/NixOS/nixpkgs/blob/2f9fd351ec37f5d479556cd48be4ca340da59b8f/pkgs/tools/inputmethods/fcitx-engines/fcitx-anthy/default.nix#L15-L18) which does the replacement for some reason.

[the remark this answers is not in the document]
Comment (lilyinstarlight) [reply pkgs/tools/inputmethods/fcitx5/fcitx5-anthy.nix:17]: If it works without it, then it's probably best to remove it and the whole `preInstall` block then

**Comment by ElnuDev** [comment]
> @ElnuDev did you test if you can input Japanese with this installed, if so we can merge this. cc @Vonfry

It works for me! @Vonfry said that they got Japanese input working as well. Thank you for the review.

**Review by Vonfry — APPROVED** [review]
LGTM

**Comment by Vonfry** [comment]
By the way, you can add fcixt5-anthy into update.py.

**Comment by ElnuDev** [comment]
> By the way, you can add fcixt5-anthy into [update.py](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/inputmethods/fcitx5/update.py).

Done!

**Review by NickCao — APPROVED** [review]
